commit 04d9a38bafddb92ab79bc0015533689e15848522
Author: Joseph Myers <joseph@codesourcery.com>
Date:   Tue Sep 1 13:47:25 2015 +0000

    Add netinet/in.h values from Linux 4.2.
    
    This patch adds new constants from Linux 4.2 to netinet/in.h:
    IPPROTO_MPLS and IP_BIND_ADDRESS_NO_PORT (both in
    include/uapi/linux/in.h in Linux; one directly in netinet/in.h, one in
    bits/in.h in glibc).
    
    Tested for x86_64 (testsuite, and that installed stripped shared
    libraries are unchanged by the patch).
    
    	* inet/netinet/in.h (IPPROTO_MPLS): New enum value and macro.
    	* sysdeps/unix/sysv/linux/bits/in.h (IP_BIND_ADDRESS_NO_PORT): New
    	macro.

diff --git a/inet/netinet/in.h b/inet/netinet/in.h
index f541c58..460745e 100644
--- a/inet/netinet/in.h
+++ b/inet/netinet/in.h
@@ -85,6 +85,8 @@ enum
 #define IPPROTO_SCTP		IPPROTO_SCTP
     IPPROTO_UDPLITE = 136, /* UDP-Lite protocol.  */
 #define IPPROTO_UDPLITE		IPPROTO_UDPLITE
+    IPPROTO_MPLS = 137,    /* MPLS in IP.  */
+#define IPPROTO_MPLS		IPPROTO_MPLS
     IPPROTO_RAW = 255,	   /* Raw IP packets.  */
 #define IPPROTO_RAW		IPPROTO_RAW
     IPPROTO_MAX
diff --git a/sysdeps/unix/sysv/linux/bits/in.h b/sysdeps/unix/sysv/linux/bits/in.h
index 887719f..fc0faf6 100644
--- a/sysdeps/unix/sysv/linux/bits/in.h
+++ b/sysdeps/unix/sysv/linux/bits/in.h
@@ -92,6 +92,7 @@
 #define IP_MINTTL       21
 #define IP_NODEFRAG     22
 #define IP_CHECKSUM     23
+#define IP_BIND_ADDRESS_NO_PORT 24
 
 /* IP_MTU_DISCOVER arguments.  */
 #define IP_PMTUDISC_DONT   0	/* Never send DF frames.  */
